frontend/js/utils/api.js

```diff
@@ -31,4 +31,56 @@ const getBattles = async () => {
   return data;
 };
 
-export { getCurrentUserData, getTeamData, getBattles };
+const createBattle = async (battle) => {
+  const data = await connectToApi(Urls.api_battle_create(), battle);
+  return data;
+};
+
+const connectToApi = (urlApi, battleData) => {
+  const tokenCSRF = getCookie('csrftoken');
+
+  if (tokenCSRF) {
+    const battleProtocol = baseUrl === 'localhost:8000' ? baseProtocolHTTP : baseProtocolHTTPS;
+
+    const urlBattle = `${battleProtocol}://${baseUrl}${urlApi}`;
+    const data = { creator: battleData.creator, opponent: battleData.opponent };
+
+    const response = axios
+      .post(urlBattle, data, { headers: { 'X-CSRFToken': tokenCSRF } })
+      .then((response) => {
+        return response;
+      })
+      .catch((error) => {
+        return handleError(error);
+      });
+    return response;
+  }
+  return null;
+};
+
+const handleError = (error) => {
+  const errorMessage = Object.values(error.response.data);
+  if (error.response.status === 400) {
+    error.response.data = { detail: errorMessage[0][0] };
+  }
+  if (error.response.status === 403) {
+    error.response.data = { detail: 'You need to be logged' };
+  }
+  return error.response;
+};
+const getCookie = (name) => {
+  let cookieValue = null;
+  if (document.cookie && document.cookie !== '') {
+    const cookies = document.cookie.split(';');
+    for (const element of cookies) {
+      const cookie = element.trim();
+      if (cookie.slice(0, Math.max(0, name.length + 1)) === `${name}=`) {
+        cookieValue = decodeURIComponent(cookie.slice(Math.max(0, name.length + 1)));
+        break;
+      }
+    }
+  }
+  return cookieValue;
+};
```

Comment:
Hmmm, how did you come to this?

Reply:
As we talked last daily meeting, I use this for getting token